rocksdb.nim

```diff
@@ -167,14 +167,19 @@ template getImpl {.dirty.} =
                        cast[cstring](unsafeAddr key[0]), key.len,
                        addr len, errors.addr)
   bailOnErrors()
-  result.ok = true
-  result.value.copyFrom(data, len)
-  # returnVal toOpenArray(cast[ptr char](data), len).to(type(result.value))
+  if not data.isNil:
+    result.ok = true
+    result.value.copyFrom(data, len)
+    rocksdb_free(data)
 
 proc get*(db: RocksDBInstance, key: KeyValueType): RocksDBResult[string] =
+  ## Get value for `key`. If no value exists, set `result.ok` to `false`,
+  ## and result.error to `""`.
```

Comment:
this is generally a bad idea for result-like return values - ie to point out what the error string will be as part of the api. if you want the user of the api to be able to react to different errors / vary their reaction, the result should hold an error enum, not a string - strings are used when all errors are "equally bad"

Reply:
so in this case, the string should really be a meaningful one - ie "key not found" or something, so as to help anyone using the api to deal with the problem without magic knowledge